CONTRIBUTING.md

```diff
@@ -52,6 +52,36 @@ The root package includes a helpful shortcut that you can use to build all packa
 pnpm validate
 ```
 
+### Filtering tests
+
+You can run tests for a specific package using a [PNPM `--filter` flag](https://pnpm.io/filtering):
+
+```sh
+pnpm --filter $PACKAGE_NAME test
+```
+
+For example to only run tests for the frames plugin:
+
+```sh
+pnpm --filter plugin-frames test
+```
+
+### Live reloading tests
+
+You can use the `test-watch` command to start a live reloading dev process that re-runs tests each time code changes. This also works with the `--filter` flag to only run tests for a specific package:
+
+```sh
+# Run tests for all packages and watch for changes:
+pnpm -r test-watch
+
+# Run tests for frames plugin and watch for changes:
+pnpm --filter plugin-frames test-watch
+```
+
+### Visualizing changes
+
+Tests output HTML files to `test/__html_snapshots__/` directories. You can open these files in a browser after running tests to see the impact of your changes.
+
```

Comment:
```suggestion

You can even combine live reloading tests (see above) with VS Code's [Live Preview](https://marketplace.visualstudio.com/items?itemName=ms-vscode.live-server) extension, which adds a `Show Preview` context menu item to all HTML files. Run the `test-watch` command, open a preview of any of the generated HTML snapshots, and the effects of your code changes will be visible in a live-updating preview right in your editor.

```

I'd suggest adding this powerful combination as well. I'm using this all the time while I'm making changes.

Reply:
Agreed — this is exactly what I ended up doing too.